src/main/java/net/neoforged/neoforge/common/crafting/RecipePriorityManager.java

```diff
@@ -0,0 +1,100 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.Resource;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.server.packs.resources.SimpleJsonResourceReloadListener;
+import net.minecraft.util.GsonHelper;
+import net.minecraft.util.profiling.ProfilerFiller;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class RecipePriorityManager extends SimpleJsonResourceReloadListener {
+    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+    public static final Logger LOGGER = LogManager.getLogger();
+
+    private Map<ResourceLocation, Integer> recipePriorities = ImmutableMap.of();
+    private static final String folder = "recipe_priorities";
+
+    public RecipePriorityManager() {
+        super(GSON, folder);
+    }
+
+    @Override
+    protected Map<ResourceLocation, JsonElement> prepare(ResourceManager resourceManager, ProfilerFiller profilerFiller) {
+        Map<ResourceLocation, JsonElement> map = super.prepare(resourceManager, profilerFiller);
+        List<ResourceLocation> finalLocations = new ArrayList<>();
+        for (String namespace : resourceManager.getNamespaces()) {
+            ResourceLocation resourceLocation = ResourceLocation.fromNamespaceAndPath(namespace, "recipe_priorities/recipe_priorities.json");
+            Optional<Resource> resource = resourceManager.getResource(resourceLocation);
+            if (resource.isPresent()) {
+                try (Reader reader = resource.get().openAsReader()) {
+                    JsonObject jsonobject = GsonHelper.fromJson(GSON, reader, JsonObject.class);
+                    boolean replace = GsonHelper.getAsBoolean(jsonobject, "replace", false);
+                    if (replace)
+                        finalLocations.clear();
+                    JsonObject entries = GsonHelper.getAsJsonObject(jsonobject, "entries");
+                    for (String entry : entries.keySet()) {
+                        ResourceLocation loc = ResourceLocation.tryParse(entry);
+                        finalLocations.remove(loc); //remove and re-add if needed, to update the ordering.
+                        finalLocations.add(loc);
+                    }
+                } catch (RuntimeException | IOException ioexception) {
+                    LOGGER.error("Couldn't read recipe priority list {} in data pack {}", resourceLocation, resource.get().sourcePackId(), ioexception);
+                }
+            }
+        }
+        Map<ResourceLocation, JsonElement> finalMap = new HashMap<>();
+        //use layered config to fetch modifier data files (modifiers missing from config are disabled)
+        for (ResourceLocation location : finalLocations) {
+            finalMap.put(location, map.get(location));
+        }
+        return map;
```

Comment:
This makes no sense, you are copying from map to final map and then returning map